middle_end/flambda/to_cmm/un_cps_closure.ml

```diff
@@ -382,7 +382,10 @@ module Greedy = struct
     (* Fill env var slots *)
     let env_map =
       Var_within_closure.Map.filter (fun clos_var _bound_to ->
-          Var_within_closure.Set.mem clos_var used_closure_vars)
+          match (used_closure_vars : _ Or_unknown.t) with
+          | Unknown -> true
+          | Known used_closure_vars ->
+            Var_within_closure.Set.mem clos_var used_closure_vars)
```

Comment:
This match may deserve to be factored out in some helper function `is_closure_var_used` somewhere

Reply:
Fixed